analytics/src/layouts/Main/Main.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { makeStyles, useTheme } from '@material-ui/styles';
+import { useMediaQuery } from '@material-ui/core';
+
+import { Sidebar, Topbar, Footer } from './components';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    paddingTop: 56,
+    height: '100%',
+    [theme.breakpoints.up('sm')]: {
+      paddingTop: 64
+    }
+  },
+  shiftContent: {
+    paddingLeft: 240
+  },
+  content: {
+    height: '100%'
+  }
+}));
+
+const Main = props => {
+  const { children } = props;
+
+  const classes = useStyles();
+  const theme = useTheme();
+  const isDesktop = useMediaQuery(theme.breakpoints.up('lg'), {
+    defaultMatches: true
+  });
+
+  const [openSidebar, setOpenSidebar] = useState(false);
```

Comment:
So we are dealing with state at component level. Could we be able to also utilise Redux in the same project? Do you foresee any need for storing application state using Redux? Any concerns or learnings?

Reply:
A medium blog discussing about React Hooks and Redux: https://medium.com/javascript-scene/do-react-hooks-replace-redux-210bab340672